research/audioset/yamnet/export.py

```diff
@@ -164,18 +176,11 @@ def make_tflite_export(weights_path, export_dir):
   # Check the TF-Lite export.
   log('Checking TF-Lite model ...')
   interpreter = tf.lite.Interpreter(tflite_model_path)
-  audio_input_index = interpreter.get_input_details()[0]['index']
-  scores_output_index = interpreter.get_output_details()[0]['index']
-  embeddings_output_index = interpreter.get_output_details()[1]['index']
-  spectrogram_output_index = interpreter.get_output_details()[2]['index']
+  runner = interpreter.get_signature_runner('serving_default')
   def run_model(waveform):
-    interpreter.resize_tensor_input(audio_input_index, [len(waveform)], strict=True)
-    interpreter.allocate_tensors()
-    interpreter.set_tensor(audio_input_index, waveform)
-    interpreter.invoke()
-    return (interpreter.get_tensor(scores_output_index),
-            interpreter.get_tensor(embeddings_output_index),
-            interpreter.get_tensor(spectrogram_output_index))
+    # Signature provides the desired dict of results with entries for
+    # 'predictions', 'embeddings', and 'log_mel_spectrogram'.
+    return runner(waveform=waveform)
   check_model(run_model, 'yamnet_class_map.csv', params)
```

Comment:
could probably just call check_model(runner, ...) and drop the redundant run_model() function altogether.

Reply:
Thanks.  I had to change the call in check_model to use a named arg (waveform=waveform) instead of positional (not entirely sure why - error was "TypeError: __call__() takes 1 positional argument but 2 were given".  But it works now.